### PR TITLE
chore(spdk-rs): add an enum to be used as channel traversal context

### PR DESCRIPTION
**Pull request by dsharma-dc**
This enum will be used to get the context about dynamic reconfiguration event happening on a device.
Based on this we'll do the minimum work required to adjust channels for this device, and not refresh for all devices.

### CODE REVIEW

File: src/io_device_traverse.rs [hunk 1]
```diff
@@ -22,6 +23,33 @@ pub enum ChannelTraverseStatus {
     Cancel,
 }
 
+#[derive(Debug, Eq, PartialEq)]
+pub enum TraverseDeviceDrEvent {
```
Comment (tiagolobocastro) [inline src/io_device_traverse.rs]: I think this should be a generic type in the api because this type probably belongs in the mayastor repo and not here.

Comment (dsharma-dc) [reply src/io_device_traverse.rs]: Yes, sure. I'll use and enhance the existing `enum DrEvent` from io-engine, and make parameter generic in `spdk-rs`.

Comment (dsharma-dc) [reply src/io_device_traverse.rs]: modified to use generics.

**Comment by dsharma-dc** [comment]
I'm trying out a change to use the existing Ctx generic type by creating a tuple with new info I need to pass. Once that is done, we'll not need this change in spdk-rs.

File: src/io_device_traverse.rs [hunk 2]
```diff
@@ -49,22 +49,27 @@ impl From<ChannelTraverseStatus> for i32 {
...
-        dyn FnMut(&mut ChannelData, &mut Ctx) -> ChannelTraverseStatus + 'a,
+        dyn FnMut(&mut ChannelData, &mut Ctx, &TrCtx) -> ChannelTraverseStatus
+            + 'a,
     >,
     /// TODO
     done_cb: Box<dyn FnMut(ChannelTraverseStatus, Ctx) + 'b>,
     /// TODO
     ctx: Ctx,
     /// TODO
+    dev_ctx: TrCtx,
```
Comment (tiagolobocastro) [inline src/io_device_traverse.rs:63]: what's the difference between Ctx and TrCtx.
We should probably document all these rather than TODO..

Comment (dsharma-dc) [reply src/io_device_traverse.rs:63]: I'm closing this PR as the mayastor repo changes(https://github.com/openebs/mayastor/pull/1469) are now using existing Ctx parameter for this purpose.
https://github.com/openebs/spdk-rs/pull/35#issuecomment-1647763070

**Comment by dsharma-dc** [comment]
I'm closing this PR as the mayastor repo changes(https://github.com/openebs/mayastor/pull/1469) are now using existing Ctx parameter for this purpose.
https://github.com/openebs/spdk-rs/pull/35#issuecomment-1647763070

**Pull request closed without merging**